### PR TITLE
fix: listView template component with id and context

### LINKED ISSUE #1125: bug: listView unexpected behaviour using template component with id and context
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description
When using a template that has a component with id and context inside, listView uses the same context reference to all itens with this id.

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Use declarative screen provided;
2. Click on first button.

## Expected Results
After clicking on the first button, only first button text should change.

## Code example, screenshot, or link to a repository:
```
private fun buildListView() = ListView(
        context = ContextData(
            id = "outsideContext",
            value = listOf("0 OUTSIDE", "1 OUTSIDE", "2 OUTSIDE", "3 OUTSIDE", "4 OUTSIDE", "5 OUTSIDE",
                "6 OUTSIDE", "7 OUTSIDE", "8 OUTSIDE", "9 OUTSIDE", "10 OUTSIDE",
                "11 OUTSIDE", "12 OUTSIDE", "13 OUTSIDE", "14 OUTSIDE", "15 OUTSIDE",
                "16 OUTSIDE", "17 OUTSIDE", "18 OUTSIDE", "19 OUTSIDE", "20 OUTSIDE")
        ),
        dataSource = expressionOf("@{outsideContext}"),
        direction = ListDirection.VERTICAL,
        template = Container(
            children = listOf(
                Text(text = expressionOf("@{item}")),
                list
            )
        ).applyStyle(
            Style(
                size = Size(width = 100.unitPercent(), height = 600.unitReal())
            )
        )
    )

    data class Person(
        val name: String,
        val cpf: Int
    )

    private val list = ListView(
        context = ContextData(
            id = "insideContext",
            value = listOf(
                Person(
                    "John",
                    0
                ),
                Person(
                    "Carter",
                    1
                ),
                Person(
                    "Josie",
                    2
                ),
                Person(
                    "Dimitri",
                    3
                ),
                Person(
                    "Maria",
                    4
                ),
                Person(
                    "Max",
                    5
                ),
                Person(
                    "Kane",
                    6
                ),
                Person(
                    "Amelia",
                    7
                ),
                Person(
                    "Jose",
                    8
                ),
                Person(
                    "Percy",
                    9
                ),
                Person(
                    "Karen",
                    10
                ),
                Person(
                    "Sol",
                    11
                ),
                Person(
                    "Jacques",
                    12
                ),
                Person(
                    "Stephen",
                    13
                ),
                Person(
                    "Sullivan",
                    14
                ),
                Person(
                    "Zoe",
                    15
                )
            )
        ),
        key = "cpf",
        dataSource = expressionOf("@{insideContext}"),
        direction = ListDirection.HORIZONTAL,
        template = Container(
            children = listOf(
                Button(
                    text = expressionOf("@{item.name} - @{item.cpf}"),
                    onPress = listOf(
                        SetContext(
                            contextId = "insideContext",
                            path = "[0].name",
                            value = "Updated John"
                        )
                    )
                ).applyStyle(
                    Style(
                        size = Size(width = 300.unitReal(), height = 80.unitReal())
                    )
                )//.setId("button")
            )
        )//.setId("container")
    ).applyStyle(
        Style(
            backgroundColor = "#CCC"
        )
    ).setId(id = "innerList")
```

### PR DESCRIPTION
**Pull request by paulomeurerzup**
Signed-off-by: paulomeurerzup <paulo.meurer@zup.com.br>

### Related Issues

This PR closes #1125 and updates the `scrollThreshold` property to `scrollEndThreshold`.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/3bd18d61474227a00dee2e76a24199a499bf5f6d/android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataManager.kt#L35">android/beagle/src/main/java/br/com/zup/beagle/android/context/ContextDataManager.kt#L35</a> - Class ‘ContextDataManager’ with ‘11’ functions detected. Defined threshold inside classes is set to ‘11’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ import android.view.View
 import android.view.ViewGroup
 import android.view.ViewGroup.LayoutParams.MATCH_PARENT
 import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import androidx.lifecycle.lifecycleScope
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt:23]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1aafdfece2ffb7b8eafd0a60e9bf536e16ac7e9a/android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt#L23"></span>The import ‘androidx.lifecycle.lifecycleScope’ is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt [hunk 2]
```diff
@@ -37,6 +38,8 @@ import br.com.zup.beagle.android.widget.WidgetView
 import br.com.zup.beagle.annotation.RegisterWidget
 import br.com.zup.beagle.core.ServerDrivenComponent
 import br.com.zup.beagle.widget.core.ListDirection
+import kotlinx.coroutines.delay
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt:41]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1aafdfece2ffb7b8eafd0a60e9bf536e16ac7e9a/android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt#L41"></span>The import ‘kotlinx.coroutines.delay’ is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt [hunk 3]
```diff
@@ -37,6 +38,8 @@ import br.com.zup.beagle.android.widget.WidgetView
...
+import kotlinx.coroutines.launch
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt:42]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1aafdfece2ffb7b8eafd0a60e9bf536e16ac7e9a/android/beagle/src/main/java/br/com/zup/beagle/android/components/ListView.kt#L42"></span>The import ‘kotlinx.coroutines.launch’ is unused.</td>
    </tr>
  </tbody>
</table>

File: android/sample/src/main/java/br/com/zup/beagle/sample/fragment/list/ListViewActivity.kt [hunk 4]
```diff
@@ -162,12 +162,12 @@ class ListViewActivity : AppCompatActivity() {
                     Style(
                         size = Size(width = 300.unitReal(), height = 80.unitReal())
                     )
-                ).setId("button")
+                )//.setId("button")
             )
-        ).setId("container")
+        )//.setId("container")
```
Comment (uziasferreirazup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/fragment/list/ListViewActivity.kt:167]: Why were these two lines commented, can we remove this snippet of code?

Comment (matheusribeirozup) [reply android/sample/src/main/java/br/com/zup/beagle/sample/fragment/list/ListViewActivity.kt:167]: I forgot to remove

**Review by uziasferreirazup — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Pull request merged**